wetterdienst/indexing/file_index_creation.py

```diff
@@ -1,69 +1,136 @@
-    Args:
-        parameter: parameter of Parameter enumeration
-        time_resolution: time resolution of TimeResolution enumeration
-        period_type: period type of PeriodType enumeration
-    Returns:
-        file index in a pandas.DataFrame with sets of parameters and station id
-    """
-    parameter_path = build_path_to_parameter(parameter, time_resolution, period_type)
 
-    files_server = list_files_of_climate_observations(parameter_path, recursive=True)
-
-    files_server = pd.DataFrame(
-        files_server, columns=[DWDMetaColumns.FILENAME.value], dtype="str"
+    file_index = _create_file_index_for_dwd_server(
+        parameter,
+        time_resolution,
+        period_type,
+        DWDCDCBase.CLIMATE_OBSERVATIONS
     )
 
-    # Filter for .zip files
-    files_server = files_server[files_server.FILENAME.str.endswith(ARCHIVE_FORMAT)]
-
-    files_server[DWDMetaColumns.FILENAME.value] = files_server[
-        DWDMetaColumns.FILENAME.value
-    ].str.replace(f"{DWD_CDC_PATH}/{DWD_CLIM_OBS_GERMANY_PATH}/", "")
+    file_index = file_index[file_index[DWDMetaColumns.FILENAME.value].str.endswith(ArchiveFormat.ZIP.value)]
 
-    files_server[DWDMetaColumns.STATION_ID.value] = files_server[
+    file_index[DWDMetaColumns.STATION_ID.value] = file_index[
         DWDMetaColumns.FILENAME.value
     ].apply(lambda x: re.findall(STATION_ID_REGEX, x)[0])
 
-    files_server.loc[:, DWDMetaColumns.STATION_ID.value] = files_server.loc[
+    file_index.loc[:, DWDMetaColumns.STATION_ID.value] = file_index.loc[
         :, DWDMetaColumns.STATION_ID.value
     ].astype(int)
 
-    files_server = files_server.sort_values(
+    file_index = file_index.sort_values(
         by=[DWDMetaColumns.STATION_ID.value, DWDMetaColumns.FILENAME.value]
     )
 
-    return files_server.loc[
+    return file_index.loc[
         :, [DWDMetaColumns.STATION_ID.value, DWDMetaColumns.FILENAME.value]
     ]
 
 
+@lru_cache
+def create_file_index_for_radolan(
+        time_resolution: TimeResolution
+) -> pd.DataFrame:
+    """
+    Function used to create a file index for the RADOLAN product. The file index will
+    include both recent as well as historical files. A datetime column is created from
+    the filenames which contain some datetime formats. This datetime column is required
+    for later filtering for the requested file.
+
+    Args:
+        time_resolution: time resolution enumeration for the requesed RADOLAN product,
+        where two are possible: hourly and daily
+
+    Returns:
+        file index as DataFrame
+    """
+    file_index = [
+        _create_file_index_for_dwd_server(
+            Parameter.RADOLAN,
+            time_resolution,
+            period_type,
+            DWDCDCBase.GRIDS_GERMANY
+        )
+        for period_type in (PeriodType.HISTORICAL, PeriodType.RECENT)
+    ]
+
+    file_index = pd.concat(file_index)
+
+    file_index = file_index[file_index[DWDMetaColumns.FILENAME.value].str.contains("/bin/")]
```

Comment:
`file_index[file_index[DWDMetaColumns.FILENAME.value].str.contains("/bin/") & file_index[DWDMetaColumns.FILENAME.value].str.endswith(
￼        (ArchiveFormat.GZ.value, ArchiveFormat.TAR_GZ.value))]`

Could be one line , or am i missing somethihg?